src/java/org/apache/cassandra/cql3/restrictions/StatementRestrictions.java

```diff
@@ -779,7 +779,12 @@ public RowFilter getRowFilter(IndexRegistry indexRegistry, QueryOptions options)
         if (filterRestrictions.isEmpty())
             return RowFilter.none();
 
-        RowFilter filter = RowFilter.create();
+        ConsistencyLevel cl = options.getConsistency();
+        boolean isStrict = cl == ConsistencyLevel.ONE || cl == ConsistencyLevel.LOCAL_ONE || cl == ConsistencyLevel.NODE_LOCAL
+                           // With zero or one restricted mutable columns, strict filtering is safe even at higher consistency levels.
+                           || nonPrimaryKeyRestrictions.getColumnDefinitions().size() <= 1;
```

Comment:
I think this is all we can do at the coordinator, before we know anything about the local data we need to search. Any other cases I'm missing?